From 55f81df0bb27c68d2a817cade5e0eb472acf24e5 Mon Sep 17 00:00:00 2001
From: Mark Asselstine <mark.asselstine@windriver.com>
Date: Wed, 14 May 2014 10:43:04 -0400
Subject: [PATCH] rdoc: fixup opt.banner heredoc

rdoc has some quirks with heredoc which in this case would
manifest in to a parse error.

| RDoc::Parser::Ruby failure around line 25 of
| lib/pry/commands/ls.rb

Fixup the heredoc to use a more standard format which is known
not to mess up rdoc processing.

Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>
---
 lib/pry/commands/ls.rb | 34 +++++++++++++++++-----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/lib/pry/commands/ls.rb b/lib/pry/commands/ls.rb
index 6ce49bb..b78f7e1 100644
--- a/lib/pry/commands/ls.rb
+++ b/lib/pry/commands/ls.rb
@@ -6,23 +6,23 @@ class Pry
     command_options :shellwords => false, :interpolate => false
 
     def options(opt)
-      opt.banner unindent <<-'BANNER'
-        Usage: ls [-m|-M|-p|-pM] [-q|-v] [-c|-i] [Object]
-               ls [-g] [-l]
-
-        ls shows you which methods, constants and variables are accessible to Pry. By
-        default it shows you the local variables defined in the current shell, and any
-        public methods or instance variables defined on the current object.
-
-        The colours used are configurable using Pry.config.ls.*_color, and the separator
-        is Pry.config.ls.separator.
-
-        Pry.config.ls.ceiling is used to hide methods defined higher up in the
-        inheritance chain, this is by default set to [Object, Module, Class] so that
-        methods defined on all Objects are omitted. The -v flag can be used to ignore
-        this setting and show all methods, while the -q can be used to set the ceiling
-        much lower and show only methods defined on the object or its direct class.
-      BANNER
+      opt.banner = <<-EOF
+Usage: ls [-m|-M|-p|-pM] [-q|-v] [-c|-i] [Object]
+       ls [-g] [-l]
+
+ls shows you which methods, constants and variables are accessible to Pry. By
+default it shows you the local variables defined in the current shell, and any
+public methods or instance variables defined on the current object.
+
+The colours used are configurable using Pry.config.ls.*_color, and the separator
+is Pry.config.ls.separator.
+
+Pry.config.ls.ceiling is used to hide methods defined higher up in the
+inheritance chain, this is by default set to [Object, Module, Class] so that
+methods defined on all Objects are omitted. The -v flag can be used to ignore
+this setting and show all methods, while the -q can be used to set the ceiling
+much lower and show only methods defined on the object or its direct class.
+  EOF
 
       opt.on :m, :methods,   "Show public methods defined on the Object (default)"
       opt.on :M, "instance-methods", "Show methods defined in a Module or Class"
-- 
1.8.3.2

